src/InteractivityComponents/Dropdown.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Automattic\WooCommerce\Blocks\InteractivityComponents;
+
+/**
+ * Dropdown class. This is a component for reuse with interactivity API.
+ *
+ * @package Automattic\WooCommerce\Blocks\InteractivityComponents
+ */
+class Dropdown {
+	/**
+	 * Render the dropdown.
+	 *
+	 * @param mixed $props The properties to render the dropdown with.
+	 * @return string|false
+	 */
+	public static function render( $props ) {
+		wp_enqueue_script( 'wc-interactivity-dropdown' );
+
+		$selected_item = $props['selected_item'] ?? array(
+			'label' => null,
+			'value' => null,
+		);
+
+		$dropdown_context = array(
+			'woocommerceDropdown' => array(
+				'selectedItem' => $selected_item,
+				'hoveredItem'  => array(
+					'label' => null,
+					'value' => null,
+				),
+				'isOpen'       => false,
+			),
+		);
+
+		$action = $props['action'] ?? '';
+
+		// TODO - translate the default.
+		$placeholder = $props['placeholder'] ?? 'Choose an option';
+
+		// Items should be an array of objects with a label and value property.
+		$items = $props['items'] ?? [];
+
+		ob_start();
+		?>
+	<div class="wc-block-stock-filter style-dropdown" data-wc-context='<?php echo wp_json_encode( $dropdown_context ); ?>' >
+		<div class="wc-blocks-components-form-token-field-wrapper single-selection" >
```

Comment:
I'm wondering if we should ditch the Form token field entirely? I mean using a new and simpler markup.

Reply:
@dinhtungdu yeah true. The approach I took here was to just migrate an existing component directly. 

Could we improve/simplify the component as a follow up? Well I don't mind either way because we can also build a new component here, it does make sense to. I just wasn't sure we want to focus effort on that right away, but I'm open to it.

Reply:
> Could we improve/simplify the component as a follow up?

Sure thing!